cub/test/catch2_test_util_device.cu

```diff
@@ -64,11 +64,16 @@ CUB_TEST("CUB correctly identifies the ptx version the kernel was compiled for",
   constexpr std::size_t single_item = 1;
   c2h::device_vector<int> cuda_arch(single_item);
 
-  int* ptx_version{};
-  cudaMallocHost(&ptx_version, sizeof(*ptx_version));
-
   // Query the arch the kernel was actually compiled for
-  get_cuda_arch_from_kernel(thrust::raw_pointer_cast(cuda_arch.data()), ptx_version);
+  int ptx_version = [&]() -> int {
+    int* buffer{};
+    cudaMallocHost(&buffer, sizeof(*buffer));
```

Comment:
Q: Do we have an RAII type for such situations? Like a `c2h::pinned_host_vector`.

Reply:
Opened #1917 to track.

Aside, it'd just be `c2h::pinned_vector`, as pinned memory is accessible from both host and device.